### DISCUSSION TITLE
Support of Blazor Component

### DISCUSSION THREAD

**Discussion by Jisu-Woniu** [giscus/giscus-component#2201] [General]
**Is your feature request related to a problem? Please describe.**
It is hard to apply giscus in Blazor WebAssembly since in Blazor components, dynamic loading of JavaScript is limited.

**Describe the solution you'd like**
A Blazor component that can be embedded in other Blazor components.

**Describe alternatives you've considered**
There is a related discussion on StackOverflow with `IJSRuntime`: <https://stackoverflow.com/questions/67560407/include-javascript-component-into-blazor-component>. It seems the second solution suggested works.
![Possible solution](https://user-images.githubusercontent.com/31986081/154954481-5e81e039-7864-44b9-b172-271229dff53b.png)

**Additional context**
I am trying to integrate it into my blog, so I will keep trying❤

**Comment by laymonage** [comment]
I'm not familiar with Blazor, so I'm not sure if I can make this happen.

In the meantime, does it support Web Components? If so, you could try [this package](https://www.npmjs.com/package/giscus).

**Comment by Jisu-Woniu** [comment]
I'm afraid that it doesn't support direct import of Web Components.

Blazor components are defined as C# classes with Razor files.

**Comment by Jisu-Woniu** [comment]
I will try using `IJSRuntime` interface as backend for giscus once I have time for it.

**Comment by Jisu-Woniu** [comment]
I wrote a wrapper here [Jisu-Woniu/giscus-blazor](https://github.com/Jisu-Woniu/giscus-blazor) and uploaded it to NuGet. I did some simple tests, but I cannot ensure they work all the time.
